### ISSUE TITLE
Some issues using logout as-is in the example

### ISSUE THREAD

**Issue by cbebry** [molekilla/hapi-passport-saml#5]
I have some interest in using hapi for a saml service provider, so I set up the examples in this repository. I was able to get the login and assert functionalities working great against an IDP (simplesamlphp) but am having some trouble with the logout functionality. 

Am I supposed to add the user object to the request object through some means? I traced through the exception trace pasted below and found that ```req.user.nameIDFormat``` in the function ```SAML.prototype.generateLogoutRequest``` expects a ```user``` object in the request that is passed in.
Do you have any thoughts on how this could be made to work? 

Thank you for your time.

```
Debug: internal, implementation, error 
    TypeError: Uncaught error: Cannot read property 'nameIDFormat' of undefined
    at SAML.generateLogoutRequest (/home/spectrum/hapi-passport-saml-example/node_modules/passport-saml/lib/passport-saml/saml.js:223:28)
    at SAML.getLogoutUrl (/home/spectrum/hapi-passport-saml-example/node_modules/passport-saml/lib/passport-saml/saml.js:435:22)
    at exports.logout (/home/spectrum/hapi-passport-saml-example/lib/v1.js:96:10)
    at Object.internals.handler (/home/spectrum/hapi-passport-saml-example/node_modules/hapi/lib/handler.js:93:36)
    at /home/spectrum/hapi-passport-saml-example/node_modules/hapi/lib/handler.js:28:23
    at internals.Protect.run (/home/spectrum/hapi-passport-saml-example/node_modules/hapi/lib/protect.js:56:5)
    at exports.execute (/home/spectrum/hapi-passport-saml-example/node_modules/hapi/lib/handler.js:22:22)
    at /home/spectrum/hapi-passport-saml-example/node_modules/hapi/lib/request.js:376:13
    at iterate (/home/spectrum/hapi-passport-saml-example/node_modules/hapi/node_modules/items/lib/index.js:35:13)
    at done (/home/spectrum/hapi-passport-saml-example/node_modules/hapi/node_modules/items/lib/index.js:27:25)
```

Example reference::
```
/**
 * Logout
 * @function
 * @param {Object} request - A Hapi Request
 * @param {Object} reply - A Hapi Reply
 */
exports.logout = function (request, reply) {
    var options = {
        name_id: request.server.app.name_id,
        session_index: request.server.app.session_index
    };

    var saml = request.server.plugins['hapi-passport-saml'].instance;

    saml.getLogoutUrl(request, function(err, url) {
        if (err !== null)
            return reply.code(500);
        return reply.redirect(url);        
    });
};
```

**Comment by molekilla** [comment]
Hi,

I think:.

1) There is room to improve this as an auth scheme, more here [Auth artifacts](http://stackoverflow.com/questions/29177093/what-are-authentication-artifacts-used-for-in-hapi-js-authentication-schemes)
2) Because we are wrapping passport-saml, you'll have to set `request.user` to `{ nameIDformat }`, it should work.

**Comment by cbebry** [comment]
If we follow the path of creating an auth scheme, how do we handle setting the credentials/artifacts objects outside of the authentication function? I attempted to set the artifacts values between routes and it does not appear that hapi is meant to be used that way.

I looked into another auth scheme, ```hapi-auth-cookie```, which appears to use state in order to save values and provides a way of managing setting values on a session, which could be used to persist the values we would want to send as part of the logout request, and likely other interactions with the auth scheme. 
That said, does it make sense to use ```hapi-auth-cookie``` for this, or is this overkill? I'd like to be able to contribute with the project so I don't mind spending some time putting together fleshed out examples, but I want to make sure I'm heading in the right direction.

To see if I could avoid going down this path, I attempted setting request.user to {nameIDFormat} but it complains about:

``` Error: Uncaught error: Missing attribute value for attribute Format of element saml:NameID```

I am planning on hardcoding the value of that attribute for now, but it seems like an auth scheme is the right direction to go, as those values initially provided by the IDP can be saved to be used in the logout request.
